src/nix-to-python.cc

```diff
@@ -4,8 +4,147 @@
 #include "internal/errors.hh"
 #include "internal/ptr.hh"
 
+#include <iostream>
+
 namespace pythonnix {
 
+PyObject *nixPosToPythonObject(nix::Pos &pos) {
+  PyObject *filename = PyUnicode_FromString(static_cast<std::string>(pos.file).c_str());
+  PyObject *line = PyLong_FromLong(pos.line);
+  PyObject *column = PyLong_FromLong(pos.column);
+  return PyTuple_Pack(3, filename, line, column);
+}
+
+PyObject *nixAttrPathToPythonObject(nix::AttrPath &path) {
+  PyObject* type = PyUnicode_FromString("ATTRPATH");
+
+  PyObject* listObj = PyList_New(path.size());
```

Comment:
Notice that for error handling I also use `PyObjPtr` to automatically decrement python objects if an error happens.

Reply:
So thank you for bringing that to my attention. I was confused why you were using `PyObjPtr` as I could not find it in `Python.h`. I see that it is a `unique_ptr` https://github.com/Mic92/pythonix/blob/master/src/internal/ptr.hh#L8-L12. What a clever way to automagically handle `Py_DECREF`.